# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/nsFontFaceLoader.cpp
# Commit: aa5527055ad1
# Full Hash: aa5527055ad1d21aa109bc8a0ceaee3b0a58288c
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-06-29 03:44:30
# Regressor Bug: 1771493
# File Overlap Count: 1
# Description:
#   Bug 1771493 - Part 1. Split FontFace(Set) into DOM facing and implementation classes. r=emilio
#   
#   This patch aims to split the implementation details of FontFace(Set)
#   into FontFace(Set)Impl classes. The Impl variants have threadsafe
#   refcounting and are intended to be the basis for worker support for
# ==============================================================================

diff -r e0c50a17b0ff -r aa5527055ad1 layout/style/nsFontFaceLoader.cpp
--- a/layout/style/nsFontFaceLoader.cpp	Tue Jun 28 15:45:58 2022 +0000
+++ b/layout/style/nsFontFaceLoader.cpp	Tue Jun 28 15:58:09 2022 +0000
@@ -45,7 +45,7 @@
 
 nsFontFaceLoader::nsFontFaceLoader(gfxUserFontEntry* aUserFontEntry,
                                    uint32_t aSrcIndex,
-                                   FontFaceSet* aFontFaceSet,
+                                   FontFaceSetImpl* aFontFaceSet,
                                    nsIChannel* aChannel)
     : mUserFontEntry(aUserFontEntry),
       mFontFaceSet(aFontFaceSet),
@@ -187,7 +187,7 @@
     nsTArray<gfxUserFontSet*> fontSets;
     ufe->GetUserFontSets(fontSets);
     for (gfxUserFontSet* fontSet : fontSets) {
-      nsPresContext* ctx = FontFaceSet::GetPresContextFor(fontSet);
+      nsPresContext* ctx = FontFaceSetImpl::GetPresContextFor(fontSet);
       if (ctx) {
         fontSet->IncrementGeneration();
         ctx->UserFontSetUpdated(ufe);
@@ -299,7 +299,7 @@
   nsTArray<gfxUserFontSet*> fontSets;
   mUserFontEntry->GetUserFontSets(fontSets);
   for (gfxUserFontSet* fontSet : fontSets) {
-    nsPresContext* ctx = FontFaceSet::GetPresContextFor(fontSet);
+    nsPresContext* ctx = FontFaceSetImpl::GetPresContextFor(fontSet);
     if (ctx) {
       // Update layout for the presence of the new font.  Since this is
       // asynchronous, reflows will coalesce.